lib/psu_identity/directory_service/client.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'oauth2'
+require 'ostruct'
+
+# @abstract Client for querying Penn State's directory-service API: https://directory-service.k8s.psu.edu/directory-service-web/resources
+module PsuIdentity::DirectoryService
+  class Error < StandardError; end
+
+  class NotFound < StandardError; end
+
+  class Client
+    attr_reader :base_url
+
+    # @param [String] base_url
+    def initialize(base_url: '/directory-service-web/resources')
+      @base_url = base_url
+    end
+
+    # @param [Hash] args of options to pass to the endpoint
+    # @option args [String] :userid of the person
+    def userid(userid)
+      process_userid_response connection.get("#{base_url}/people/#{userid}")
+    end
+
+    private
+
+      def process_userid_response(response)
+        raise NotFound if response.status == 404
+
+        raise Error.new(response.body) unless response.success?
+
+        snake_cased_response = JSON.parse(response.body).transform_keys(&:underscore)
+        OpenStruct.new(snake_cased_response)
```

Comment:
We might eventually want to have a person class if we need to add custom methods, but this seems fine for now, IMO.

Reply:
I kept going back and forth on this, The Person class got gnarly real quick, and if this gem is intended to be a wrapper on PSU's apis, it might seem odd to return something mutated from what is in central's database?

Reply:
^just explaining my motivation, i'm prepared to be wrong.